pkg/gardenlet/controller/shoot/shoot/reconciler_delete.go

```diff
@@ -119,6 +120,9 @@ func (r *Reconciler) runDeleteShootFlow(ctx context.Context, o *operation.Operat
 			return nil
 		}),
 		errors.ToExecute("Retrieve the infrastructure resource", func() error {
+			if isWorkerless {
```

Comment:
any reason to check this inside the fn that is passed to `ToExecute`. You could have had an outer condition:
```go
if !isWorkerless {
  errors.ToExecute("Retrieve the infrastructure resource", func() error {
     }
  }
}
```

Reply:
`errors.ToExecute` is a task function, we cannot add the condition outside

Reply:
using closure you can. I looked at the code again and `runDeleteShootFlow` is a extremely log method.

Reply:
`using closure you can.`
Can you please explain?